src/main/java/Duke.java

```diff
@@ -1,10 +1,259 @@
+    private static final String MESSAGE_TASK_COMPLETED = "marked as completed, well done!";
+    private static final String MESSAGE_ALL_TASK_COMPLETED = "All Task Completed!";
+    private static final String MESSAGE_TASKS_LEFT = "You a total of tasks remaining: ";
+    private static final String MESSAGE_NO_TASKS_DUE = "Fortunately, you have no tasks due";
+    private static final String MESSAGE_GOODBYE = "Bye! Hope to see you again soon!";
+    private static final String MESSAGE_GREETING = "Hello! I'm Duke!" + System.lineSeparator()
+            + "What can I do for you?";
+    private static final String MESSAGE_INVALID_OPTION = "Invalid option, Try again!"
+            + System.lineSeparator();
+
+    /**
+     * Main entry point of the application
+     * Greets the User and begins User interaction
+     */
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        printWelcomeScreen();
+        //Interaction Component
+        boolean stillInteracting = true;
+        while (stillInteracting) {
+            String[] processedInputs = processInput(getUserInput());
+            stillInteracting = executeCommand(processedInputs);
+        }
+    }
+
+    /**
+     * Returns the raw user input into processed parts for execution based on found keywords
+     * @param userInput raw input from user
+     * @return processedInputs String array with the input in parts as follows
+     * processedInputs[0] the processed Command e.g COMMAND_DONE, COMMAND_TODO, ...
+     * processedInputs[1] taskName e.g 'borrow book' or 1 (taskID for done commands)
+     * processedInputs[2] = taskParameter, e.g 'Sunday', 'Mon 2pm-4pm'
+     **/
+    private static String[] processInput(String userInput) {
+        String[] inputParts = userInput.trim().split(RAW_COMMAND_DELIMIT, SPLIT_INPUT_LIMIT);
+        switch (inputParts[0]) {
+        case COMMAND_DONE:
+            return splitInputToParts(inputParts, DELIMIT_DONE, COMMAND_DONE);
+        case COMMAND_TODO:
+            return splitInputToParts(inputParts, DELIMIT_TODO, COMMAND_TODO);
+        case COMMAND_DEADLINE:
+            return splitInputToParts(inputParts, DELIMIT_DEADLINE, COMMAND_DEADLINE);
+        case COMMAND_EVENT:
+            return splitInputToParts(inputParts, DELIMIT_EVENT, COMMAND_EVENT);
+        default:
+            //Catch List and Bye commands that need no processing
+            return new String[]{ inputParts[0], INIT_STRING, INIT_STRING };
+        }
+    }
+
+    /**
+     * Returns the split inputParts into 3 distinct data: Command, Task name, Task Parameters
+     * @param inputParts Separated user input into 2 parts by the first spacing
```

Comment:
Should there be an empty line left in between the description and parameter section? I noticed the same issue in other places too. 
Perhaps you could also add a punctuation behind each parameter description.

Reply:
Thank you for your review, I have resolved the issue!